CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -460,69 +460,18 @@ function Humanoid:getCurrentMood()
   end
 end
 
-local function Humanoid_startAction(self)
+--! Start the next (always first) action in the queue.
+function Humanoid:startAction()
   local action = self.action_queue[1]
 
   -- Handle an empty action queue in some way instead of crashing.
   if not action then
-    -- if this is a patient that is going home, an empty
-    -- action queue is not a problem
-    if class.is(self, Patient) and self.going_home then
-      return
-    end
+    self:handleEmptyActionQueue() -- Inserts an action into the action queue.
```

Comment:
```suggestion
    self:_handleEmptyActionQueue() -- Inserts an action into the action queue.
```

Reply:
Yep

Reply:
made the empty action queue handler private